File: lintcode/easy/length_of_last_word/py/length_of_last_word.py
# coding:utf-8
'''
@Copyright:LintCode
@Author:   lilsweetcaligula
@Problem:  http://www.lintcode.com/problem/length-of-last-word
@Language: Python
@Datetime: 17-02-15 13:57
'''

class Solution:
    # @param {string} s A string
    # @return {int} the length of last word
    def lengthOfLastWord(self, s):
        words = s.split()
        
        if len(words) == 0:
            return 0
        
        return len(words[-1])

File: lintcode/easy/length_of_last_word/cpp/length_of_last_word.cpp
/*
@Copyright:LintCode
@Author:   lilsweetcaligula
@Problem:  http://www.lintcode.com/problem/length-of-last-word
@Language: C++
@Datetime: 17-02-15 16:14
*/

#include <vector>
#include <sstream>

class Solution {
public:
    /**
     * @param s A string
     * @return the length of last word
     */
    int lengthOfLastWord(std::string& s) 
    {
        std::vector<std::string> words;
        std::istringstream iss { s };
        std::string word;
    
        while (iss >> word) {
            words.push_back(word);
        }
            
        if (words.size() == 0) {
            return 0;
        }
            
        return words.back().size();
    }
};